src/screens/Forum/ForumPostScreen.js

```diff
@@ -1,10 +1,127 @@
+            setPost(
+              <Card>
+                <Card.Title title={postData.title} subtitle={`${authorData.displayName} ${postData.createdAt.toDate().toLocaleTimeString('en-US')}`} left={(props) => <Avatar.Text {...props} label={authorData.displayName.charAt(0).toUpperCase()} />} />
+                <Card.Content>
+                  <Paragraph>{postData.body}</Paragraph>
+                </Card.Content>
+                <Card.Actions>
+                  <Button>Reply</Button>
+                </Card.Actions>
+              </Card>,
+            );
+
+            /* Query for post comment data */
+            firestore().collection('forum_comments').where('postID', '==', postId)
+              .get()
+              .then((snapshot) => {
+                const repliesData = snapshot.docs.map((doc) => doc.data());
+
+                /* Sort comments by created at timestamp, from oldest to youngest */
+                const sortedRepliesData = repliesData.sort((a, b) => {
+                  const aMillis = a.createdAt.toMillis();
+                  const bMillis = b.createdAt.toMillis();
+                  if (aMillis > bMillis) {
+                    return 1;
+                  }
+                  if (aMillis < bMillis) {
+                    return -1;
+                  }
+                  return 0;
+                });
+
+                /* Query comment author user data */
+                const userSnapshot = sortedRepliesData.map((reply) => firestore().collection('users').doc(reply.userID).get());
+
+                Promise.all(userSnapshot).then((userSnap) => {
+                  const usersData = userSnap.map((doc) => doc.data());
+
+                  /* Set replies hook with comment and comment author data */
+                  setReplies(sortedRepliesData.map((reply, i) => (
+                    <Card>
+                      <Card.Title subtitle={`${usersData[i].displayName} ${reply.createdAt.toDate().toLocaleTimeString('en-US')}`} />
+                      <Card.Content>
+                        <Paragraph>{reply.body}</Paragraph>
+                      </Card.Content>
+                    </Card>
+                  )));
+                });
+              })
+              .catch((error) => {
+                setErrorMessage(error.message);
+              });
+          }
+        });
+      })
+      .catch((error) => {
+        setErrorMessage(error.message);
+      });
+  }, [postId]);
+
+  const onRefresh = useCallback(() => {
+    setRefreshing(true);
+    getData();
+    wait(2000).then(() => setRefreshing(false));
+  }, [getData]);
+
+  useEffect(() => {
+    getData();
+  }, [getData, postId]);
 
-export default function ForumPostScreen() {
   return (
     <View>
-      <Text>Not implemented</Text>
+      <ScrollView refreshControl={
+        <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
+        }
+      >
+        {errorMessage && <Text>{errorMessage}</Text>}
+        {post}
```

Comment:
Can you add a loading state and an `<ActivityIndicator />` that shows when the post and comments are loading? See twitterer for an example.